app/lib/event_store/active_record_event_repository.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module EventStore
+  class ActiveRecordEventRepository < RailsEventStoreActiveRecord::EventRepository
```

Comment:
Well, the context is in the comment below:
>

This is the main change, plus there is a slight refactor. The original method implementation is:
```
    def build_event_entity(record)
      return nil unless record
      record.event_type.constantize.new(
        record.data.merge(
          event_id: record.event_id,
          metadata: record.metadata
        )
      )
    end
```

However, now that I'm thinking... I also changed the initializer of the `BaseEventStoreEvent`, maybe that was actually enough?... (but then probably I wouldn't have added this piece) - I'll take another look.

Reply:
Understood 👍

Reply:
OK, I checked, and it turns out that overriden `initialize` method on `BaseEventStoreEvent` only fixes the cases where this class or its subclasses are used.

However, in many places in the tests we use the superclass `RailsEventStore::Event`, provided by the rails event store library. 

For example:
```
    event_data = repository.adapter.create!(stream: 'foo', event_type: RailsEventStore::Event,
                                            event_id: SecureRandom.uuid, data: {}, metadata: {})

    assert event = Repository.find_event(event_data.event_id)
```
It's also used in the event factory.

We could replace all usages with `BaseEventStoreEvent` throughout the code, but I'm not sure whether it would actually be better. I think this solution is somewhat "safer", because it still allows us to use the pure library's class `RailsEventStore::Event` instead of `BaseEventStoreEvent`.